File: Logic_HandsOn/Java/Problems on Array,Pattern etc/Program79.java
////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
//
//	Program:
//		Accept N numbers from user and accept Range, Display all elements from that range
//
////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////


import java.util.*;

class Solution
{
	public int arr[],iCnt,iStart,iEnd;

	public Solution(int iNo,int iStart,int iEnd)
	{
		this.iStart=iStart;
		this.iEnd=iEnd;
		this.iCnt = 0;
		arr = new int[iNo];
	}

	public void Accept()
	{
		Scanner sObj = new Scanner(System.in);

		System.out.println("Enter "+arr.length+" elements: ");
		for(iCnt=0;iCnt<arr.length;iCnt++)
		{
			System.out.print("Enter "+ (iCnt+1) + "element: ");
			arr[iCnt] = sObj.nextInt();
		}
	}	

	public void Range()
	{
		for(iCnt=0;iCnt<arr.length;iCnt++)
		{
			if(arr[iCnt]>=iStart && arr[iCnt]<=iEnd)
			{
				System.out.println(arr[iCnt]);
			}
		}

	}

}

class Program4
{
	public static void main(String[] args)
	{
		Scanner sObj = new Scanner(System.in);

		System.out.println("Enter number of elements:");
		int iLength = sObj.nextInt();
		
		System.out.println("Enter starting point:");
		int iStart = sObj.nextInt();

		System.out.println("Enter ending point:");
		int iEnd = sObj.nextInt();
		
		Solution obj = new Solution(iLength,iStart,iEnd);
		obj.Accept();
		obj.Range();
	
	}
}

/*
OUTPUT:
Enter number of elements:
6
Enter starting point:
60
Enter ending point:
90
Enter 6 elements: 
Enter 1element: 85
Enter 2element: 66
Enter 3element: 3
Enter 4element: 76
Enter 5element: 93
Enter 6element: 88
85 
66 
76 
88 



*/

File: Logic_HandsOn/C++/Problems on Digits,String,Array,Pointers etc/program79.cpp
////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
//
//	Program:
//		Accept N numbers from user and accept Range, Display all elements from that range
//
////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////


#include<iostream>
using namespace std;

class Solution
{
	private:
		int *iArr;
		int iSize,iCnt,iStart,iEnd;

	public:
		Solution(int iNo=0,int iStart=0,int iEnd=0,int iCnt=0)
		{
			this->iStart=iStart;
			this->iEnd=iEnd;
			this->iSize=iNo;
			iArr = new int[iSize];
		}

		void Accept()
		{
			cout<<"Enter "<<iSize<<" elements:\n";
			for(iCnt=0;iCnt<iSize;iCnt++)
			{
				cout<<"Enter "<<iCnt+1<<" element: ";
				cin>>iArr[iCnt];
			}
		}
		
		void Range()
		{
			for(iCnt=0;iCnt<iSize;iCnt++)
			{
				if(iArr[iCnt]>=iStart && iArr[iCnt]<=iEnd)
				{
					cout<<iArr[iCnt]<<" ";
				}
				
			}

			cout<<"\n";
		}

		~Solution()
		{
			delete []iArr;
		}

};

int main()
{
	int iLength=0,iRet=0,iStart=0,iEnd=0;

	cout<<"Enter number of elements: ";
	cin>>iLength;
	
	cout<<"Enter starting point: ";
	cin>>iStart;

	cout<<"Enter ending point: ";
	cin>>iEnd;
	
	Solution *dObj = new Solution(iLength,iStart,iEnd);

	dObj->Accept();
	dObj->Range();

	delete dObj;
	return 0;
}

/*
OUTPUT:
Enter number of elements: 6
Enter starting point: 60
Enter ending point: 90
Enter 6 elements:
Enter 1 element: 85
Enter 2 element: 66
Enter 3 element: 3
Enter 4 element: 76
Enter 5 element: 93
Enter 6 element: 88
85 66 76 88 

Enter number of elements: 6
Enter starting point: 30
Enter ending point: 50
Enter 6 elements:
Enter 1 element: 85
Enter 2 element: 66
Enter 3 element: 3
Enter 4 element: 76
Enter 5 element: 93
Enter 6 element: 88


*/
